octopusdeploy/schema_team.go

```diff
@@ -146,6 +146,64 @@ func getTeamSchema() map[string]*schema.Schema {
 			Optional:    true,
 			Type:        schema.TypeList,
 		},
+		"user_role": {
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"environment_ids": {
+						ConfigMode: schema.SchemaConfigModeAttr,
+						Elem: &schema.Schema{
+							Type: schema.TypeString,
+						},
+						Optional: true,
+						Set:      schema.HashString,
+						Type:     schema.TypeSet,
+					},
+					"id": {
+						Computed: true,
+						Type:     schema.TypeString,
+					},
+					"project_ids": {
+						ConfigMode: schema.SchemaConfigModeAttr,
+						Elem: &schema.Schema{
+							Type: schema.TypeString,
+						},
+						Optional: true,
+						Set:      schema.HashString,
+						Type:     schema.TypeSet,
+					},
+					"project_group_ids": {
+						ConfigMode: schema.SchemaConfigModeAttr,
+						Elem: &schema.Schema{
+							Type: schema.TypeString,
+						},
+						Optional: true,
+						Set:      schema.HashString,
+						Type:     schema.TypeSet,
+					},
+					"space_id": {
+						Type:     schema.TypeString,
+						Required: true,
+					},
+					"tenant_ids": {
+						ConfigMode: schema.SchemaConfigModeAttr,
+						Elem: &schema.Schema{
+							Type: schema.TypeString,
+						},
+						Optional: true,
+						Set:      schema.HashString,
+						Type:     schema.TypeSet,
+					},
+					"user_role_id": {
+						Type:     schema.TypeString,
```

Comment:
In UI we can add multiple roles in team, here we are are giving access to single user role id.

Reply:
The "user_role" property of the team is a Set. Terraform for it would look like

```terraform
resource "octopusdeploy_team" "this" {
  name = "Example"

  user_role {
    user_role_id = "userrole-1"
    space_id     = "Spaces-1"
  }
  user_role {
    user_role_id = "userrole-2"
    space_id     = "Spaces-1"
  }
}
```

Reply:
When this will be available?

Reply:
@ajaychoudhary-bcg I'll try to have this PR ready for review by the end of the week. I've been doing these changes in my free time. Everything is merged on the api library side so I need to rebase with upstream and write tests.

Reply:
@evan-cleary Please merge and create new version so that we can use it.